monerochad/client.py

```diff
@@ -49,20 +49,31 @@ async def on_ready(self):
 			logger.info("Feed loop started")
 		else:
 			logger.info("Skipping feed loop start")
-	
-	@tasks.loop(seconds=settings.NICK_LOOP_INTERVAL_SECONDS)
-	async def nick_loop(self):
-		try:
-			cp = coingecko.get_price("monero")
-			nick = f"${cp.usd:.2f} | MoneroChad"
-			for guild in self.guilds:
-				await guild.me.edit(nick=nick)
-		except BaseException as ex:
-			logger.error("Exception in the nick loop", exc_info=True)
-			for da_id in debug_admin_ids:
-				da = await self.fetch_user(da_id)
-				await da.send(f"Exception in the news feed loop:\n```\n{ex}\n```")
-	
+
+        @tasks.loop(seconds=settings.NICK_LOOP_INTERVAL_SECONDS)
+        async def nick_loop(self):
+            try:
+                # main loop
+                cp = coingecko.get_price("monero")
+                nick = f"${cp.usd:.2f} | MoneroChad"
+                for guild in self.guilds:
+                    try:
+                        await guild.me.edit(nick=nick)
+                    except discord.errors.Forbidden as forbidden_error:
+                        logger.error(f"Could not change nickname in guild {guild.id}: {forbidden_error}")
+                        # even if it fails, continue with other guilds
+            except BaseException as ex:
+                logger.error("Exception in the nick loop", exc_info=True)
+                for da_id in debug_admin_ids:
+                    try:
+                        da = await self.fetch_user(da_id)
+                        await da.send(f"Exception in the nick loop:\n```\n{ex}\n```")
+                    except discord.errors.Forbidden as forbidden_error: # God forbid not send message, god teach not interrupt dragon tiger fight
```

Comment:
I don't get the comment lol
But if we're trying to catch errors to prevent the nick loop from dying, we should probably catch all of them, not just `Forbidden`.

Reply:
There can only be 1 major error here, thats the forbidden thingy.

Reply:
as i mentioned that's not true, the bot will never get a forbidden error. what makes the current version of the bot fail is temporary network connection issues, which is what ought be caught here. for good measure i'd just catch any `Exception` as the ones thrown on connection issues don't seem to be well documented.

Reply:
alright but the try block is just fetching user and sending msg to him so a connection issue is quite weird